### DISCUSSION TITLE
The selection of observation type

### DISCUSSION THREAD

**Discussion by Exasor** [eleurent/highway-env#159] [General]
Hi, I found that in merge_env, the observation type is TimeToCollision and in highway_env it is Kinematics, so I wonder if I can use Kinematics in merge_env, and how to choose the proper observation type in different env?

**Comment by Exasor** [comment]
And in highway env, the speed reward is related to the reward_speed_range
`scaled_speed = utils.lmap(self.vehicle.speed, self.config["reward_speed_range"], [0, 1])`
but in merge env, the `speed_index` and `SPEED_COUNT` are used to compute speed reward instead, I don't know why to make this change
`self.HIGH_SPEED_REWARD * self.vehicle.speed_index / (self.vehicle.SPEED_COUNT - 1)`

Indeed, I'm not very clear about the reason to use the index-speed transformation method in the speed control of MDPVehicle and why SPEED_COUNT value is fixed to 3? Sincerely looking for your help, thanks a lot!

**Comment by eleurent** [comment]
Hi,

> Hi, I found that in merge_env, the observation type is TimeToCollision and in highway_env it is Kinematics, so I wonder if I can use Kinematics in merge_env, and how to choose the proper observation type in different env?

Yes, you can choose any type of observation, see [the documentation](https://highway-env.readthedocs.io/en/latest/observations/index.html)

> Indeed, I'm not very clear about the reason to use the index-speed transformation method in the speed control of MDPVehicle and why SPEED_COUNT value is fixed to 3? Sincerely looking for your help, thanks a lot!

The SPEED_COUNT refers to the number of different target speeds that can be chosen by the FASTER and SLOWER actions, between SPEED_MIN and SPEED_MAX. For instance, for min=20, max=30 and count=3, you can select a desired speed in {20, 25, 30}. In the past, I used the index {0, 1, 2} of these speeds as a proxy to get the ratio of current speed and maximum speed.
But in some environments (like HighwayEnv) I changed it to a continuous map. 
But I agree that the current situation is not ideal, this reward term should be consistent between environments.

Does that help?

**Comment by Exasor** [comment]
Yes, thanks a lot for your reply
